Files/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -795,6 +856,58 @@ public async Task<ReturnResult> RenameAsync(IStorageItemWithPath source, string
 
         #endregion IFilesystemHelpers
 
+        private static async Task<(NameCollisionOption collision, bool cancelOperation)> GetCollision(FilesystemOperationType operationType, IEnumerable<IStorageItemWithPath> source, IEnumerable<string> destination, bool forceDialog)
+        {
+            List<FilesystemItemsOperationItemModel> incomingItems = new List<FilesystemItemsOperationItemModel>();
+            List<FilesystemItemsOperationItemModel> conflictingItems = new List<FilesystemItemsOperationItemModel>();
+
+            NameCollisionOption collision = NameCollisionOption.FailIfExists;
+
+            for (int i = 0; i < source.Count(); i++)
+            {
+                incomingItems.Add(new FilesystemItemsOperationItemModel(operationType, source.ElementAt(i).Path ?? source.ElementAt(i).Item.Path, destination.ElementAt(i)));
+
+                // TODO: Using StorageItemHelpers.Exists() on large amount of items is slow. Replace it with something faster
+                if (destination.Count() > 0 && await StorageItemHelpers.Exists(destination.ElementAt(i))) // Same item names in both directories
```

Comment:
Checking whether file exists is slow on large amount of items. Unfortunately I didn't came up with anything better

Reply:
I guess the alternative would be to just try to perform the operation _without_ checking if the destination exist. If the operation fails show the conflict dialog.

Rough pseudo-code below:
```cs
var conflictingItems = new List<IStorageItemWithPath>();

for (int i = 0; i < source.Count(); i++)
{
    var resultCode = await filesystemOperations.CopyAsync(
        source.ElementAt(i),
        destination.ElementAt(i));
    if (resultCode == FileSystemStatusCode.AlreadyExists)
    {
        conflictingItems.Add(source.ElementAt(i));
    }
}

if (conflictingItems.Any())
{
    // Show conflicts dialog and repeat operation
}
```

Reply:
That'd require a lot of changes to current logic. I want to keep it as is - show all items copied and conflicts before copy/cut is initiated. What do you say about calling the FullTrust process to do bulk check on files, is this a good idea? @gave92

Reply:
I agree the above is a change that can be done in a future PR.

> What do you say about calling the FullTrust process to do bulk check on files

I wouldn't do that, but we could check if using native operations like the following leads to better performance?

```cs
public static bool Exists(string path, IShellPage associatedInstance = null)
{
    return NativeFileOperationsHelper.GetFileAttributesExFromApp(
        path, NativeFileOperationsHelper.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, out _);
}
```

Reply:
> ```cs
> public static bool Exists(string path, IShellPage associatedInstance = null)
> {
>     return NativeFileOperationsHelper.GetFileAttributesExFromApp(
>         path, NativeFileOperationsHelper.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, out _);
> }
> ```
:eyes:

Reply:
Thanks @gave92!

Reply:
It's a bit faster now